sdk/samples/iot/sample.c

```diff
@@ -0,0 +1,454 @@
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      ENV_IOT_DEVICE_X509_TRUST_PEM_FILE_PATH, "", false, &(env_vars->x509_trust_pem_file_path)));
+
+  LOG(" "); // Log formatting
+  return AZ_OK;
+}
+
+az_result read_configuration_entry(
+    const char* env_name,
+    char* default_value,
+    bool hide_value,
+    az_span* out_value)
+{
+  PRECONDITION_NOT_NULL(env_name);
+  PRECONDITION_NOT_NULL(out_value);
+
+  char* env_value = getenv(env_name);
+
+  if (env_value == NULL && default_value != NULL)
+  {
+    env_value = default_value;
+  }
+
+  if (env_value != NULL)
+  {
+    (void)printf("%s = %s\n", env_name, hide_value ? "***" : env_value);
+    az_span env_span = az_span_create_from_str(env_value);
+
+    // assumes *out_value is a valid span
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(*out_value, az_span_size(env_span));
+    az_span_copy(*out_value, env_span);
+    *out_value = az_span_slice(
+        *out_value, 0, az_span_size(env_span)); // Trim any excess. Important for numbers
+  }
+  else
+  {
+    LOG_ERROR("(missing) Please set the %s environment variable.", env_name);
+    return AZ_ERROR_ARG;
+  }
+
+  return AZ_OK;
+}
+
+az_result create_mqtt_endpoint(
+    sample_type type,
+    const sample_environment_variables* env_vars,
+    char* endpoint,
+    size_t endpoint_size)
+{
+  PRECONDITION_NOT_NULL(env_vars);
+  PRECONDITION_NOT_NULL(endpoint);
+
+  if (type == PAHO_IOT_HUB)
+  {
+    int32_t required_size = az_span_size(mqtt_url_prefix) + az_span_size(env_vars->hub_hostname)
+        + az_span_size(mqtt_url_suffix) + (int32_t)sizeof('\0');
+
+    if (required_size > (int32_t)endpoint_size)
+    {
+      return AZ_ERROR_INSUFFICIENT_SPAN_SIZE;
+    }
+
+    az_span hub_mqtt_endpoint = az_span_create((uint8_t*)endpoint, (int32_t)endpoint_size);
+    az_span remainder = az_span_copy(hub_mqtt_endpoint, mqtt_url_prefix);
+    remainder = az_span_copy(remainder, env_vars->hub_hostname);
+    remainder = az_span_copy(remainder, mqtt_url_suffix);
+    az_span_copy_u8(remainder, '\0');
+  }
+  else if (type == PAHO_IOT_PROVISIONING)
+  {
+    int32_t required_size = az_span_size(provisioning_global_endpoint) + (int32_t)sizeof('\0');
+
+    if (required_size > (int32_t)endpoint_size)
+    {
+      return AZ_ERROR_INSUFFICIENT_SPAN_SIZE;
+    }
+
+    az_span provisioning_mqtt_endpoint = az_span_create((uint8_t*)endpoint, (int32_t)endpoint_size);
+    az_span remainder = az_span_copy(provisioning_mqtt_endpoint, provisioning_global_endpoint);
+    az_span_copy_u8(remainder, '\0');
+    ;
+  }
+  else
+  {
+    LOG_ERROR("Sample type undefined.");
+    return AZ_ERROR_ARG;
+  }
+
+  LOG_SUCCESS("MQTT endpoint created at \"%s\".", endpoint);
+
+  return AZ_OK;
+}
+
+void sleep_for_seconds(uint32_t seconds)
+{
+#ifdef _WIN32
+  Sleep((DWORD)seconds * 1000);
+#else
+  sleep(seconds);
+#endif
+  return;
+}
+
+uint32_t get_epoch_expiration_time_from_minutes(uint32_t minutes)
+{
+  return (uint32_t)(time(NULL) + minutes * 60 * 60);

```

Comment:
?? time returns seconds

```suggestion
  return (uint32_t)(time(NULL) + minutes * 60);

```